pkg/api/pluginproxy/ds_proxy.go

```diff
@@ -272,6 +272,10 @@ func (proxy *DataSourceProxy) director(req *http.Request) {
 	if proxy.features.IsEnabled(featuremgmt.FlagIdForwarding) {
 		proxyutil.ApplyForwardIDHeader(req, proxy.ctx.SignedInUser)
 	}
+
+	if proxy.features.IsEnabled(featuremgmt.FlagTeamHeaders) {
+		proxyutil.ApplyTeamHeaders(req, proxy.ds, proxy.ctx.Teams)
```

Comment:
This probably won't work for Loki/other core datasources since it make its requests from the backend I am pretty sure https://github.com/grafana/grafana/blob/main/pkg/tsdb/loki/api.go

Probably need to add it here to apply to both proxy and backend plugins/requests: https://github.com/grafana/grafana/blob/625b400fafbdcd484aaf8622a829e493d3bd94ba/pkg/infra/httpclient/httpclientprovider/http_client_provider.go#L29

Reply:
@alexanderzobnin was this the part you mentioned about the `X-ID-Token`?

if so; could we add that as the next PR?